resources/services/athena/data_catalogs.go

```diff
@@ -0,0 +1,306 @@
+										Resolver:    schema.ParentIdResolver,
+									},
+									{
+										Name:        "name",
+										Description: "The name of the column",
+										Type:        schema.TypeString,
+									},
+									{
+										Name:        "comment",
+										Description: "Optional information about the column",
+										Type:        schema.TypeString,
+									},
+									{
+										Name:        "type",
+										Description: "The data type of the column",
+										Type:        schema.TypeString,
+									},
+								},
+							},
+							{
+								Name:        "aws_athena_data_catalog_database_table_partition_keys",
+								Description: "Contains metadata for a column in a table",
+								Resolver:    fetchAthenaDataCatalogDatabaseTablePartitionKeys,
+								Columns: []schema.Column{
+									{
+										Name:        "data_catalog_database_table_cq_id",
+										Description: "Unique CloudQuery ID of aws_athena_data_catalog_database_tables table (FK)",
+										Type:        schema.TypeUUID,
+										Resolver:    schema.ParentIdResolver,
+									},
+									{
+										Name:        "name",
+										Description: "The name of the column",
+										Type:        schema.TypeString,
+									},
+									{
+										Name:        "comment",
+										Description: "Optional information about the column",
+										Type:        schema.TypeString,
+									},
+									{
+										Name:        "type",
+										Description: "The data type of the column",
+										Type:        schema.TypeString,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+// ====================================================================================================================
+//                                               Table Resolver Functions
+// ====================================================================================================================
+
+func fetchAthenaDataCatalogs(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- interface{}) error {
+	c := meta.(*client.Client)
+	svc := c.Services().Athena
+	input := athena.ListDataCatalogsInput{}
+	for {
+		response, err := svc.ListDataCatalogs(ctx, &input, func(options *athena.Options) {
+			options.Region = c.Region
+		})
+		if err != nil {
+			return diag.WrapError(err)
+		}
+		for _, d := range response.DataCatalogsSummary {
+			dc, err := svc.GetDataCatalog(ctx, &athena.GetDataCatalogInput{
+				Name: d.CatalogName,
+			}, func(options *athena.Options) {
+				options.Region = c.Region
+			})
+			if err != nil {
+				// retrieving of default data catalog (AwsDataCatalog) returns "not found error" but it exists and its
+				// relations can be fetched by its name
+				if *d.CatalogName == "AwsDataCatalog" {
+					res <- types.DataCatalog{Name: d.CatalogName, Type: d.Type}
+					continue
+				}
+				return diag.WrapError(err)
+			}
+			res <- *dc.DataCatalog
+		}
```

Comment:
This should be done in a Go Routine like this:
https://github.com/cloudquery/cq-provider-aws/blob/5cd40bd911b6cd0bbc2c14c4edb8726e76e1b5ad/resources/services/ecs/task_definitions.go#L658-L674